stw_potsdam/feed.py

```diff
@@ -48,18 +51,23 @@ def _process_day(builder, day):
                         roles=None)
 
 
-def _create_builder(canteen):
+def render_menu(canteen, menu):
     builder = LazyBuilder()
+
+    for day in _active_days(menu):
+        _process_day(builder, day)
+
+    return builder.toXMLFeed()
+
+
+def render_meta(canteen):
```

Comment:
Your current approach looks very neat, I'm that close to approving. Main culprit: tests do not pass due errors in argument passing. Second, some things appear half-finished to me in `feed.py`:

* Remove unused import statement for `json`. (Yes, at some time I should really think about integrating basic automatic static analysis / linting)
* `render_menu` no longer needs a canteen parameter.
* `render_meta` still references `reverse` function from ex-module `reverse`, which rightfully ceased to exist. It should take the canteen menu URL by parameter, populated via `url_for` inside the `views`module.

I estimate a required fix time of two minutes 😉

Reply:
Whoa, I see that I should not produce code when I'm in a hurry. I thought that all the tests passed (and I was surprised that everything worked out in the beginning), but it turns out that my test runner used the cached Docker image and not the current source code. 🤦‍♂️ 
Fixes should be up any minute.